client-lib/java/pom.xml

```diff
@@ -5,7 +5,7 @@
   <groupId>com.abcxyz.jvs</groupId>
   <artifactId>jvs-client</artifactId>
   <packaging>jar</packaging>
-  <version>0.0.1-SNAPSHOT</version>
+  <version>${revision}</version>
```

Comment:
is this the git revision?

Reply:
Right. We will create git tags like `v0.0.1` and the release version will be that without the `v`.
For ref: https://maven.apache.org/maven-ci-friendly.html

Reply:
That link seems to recommend the opposite - that it should be specific in the pom file. 

```
But wait there is a problem? Which version will the artifacts have? So you need to define the version for your artifacts. The first possibility is to use the command line like this:

mvn -Drevision=1.0.0-SNAPSHOT clean package

This will become cumbersome over the time. So the other solution for this is to simply use a property inside the pom file which looks like this:


<properties>
    <revision>1.0.0-SNAPSHOT</revision>
  </properties>
```

Reply:
IIUC, the problem is about _not_ having `${revision}` being a property with a default value. But I've added it as a property.

Reply:
Can we pull the VCS information from git instead of requiring a user to supply it?

Reply:
I didn't find a native mvn way to auto pull in VCS info. My idea is to run `mvn clean deploy -Drevision=[VERSION]` in the release workflow where we would pass the git tag to `-Drevision`.